tests/validation/validation_data.cpp

```diff
@@ -15,6 +16,38 @@ namespace mc {
 
 trace_io g_trace_io;
 
+util::path trace_io::find_datadir() {
+    // If environment variable is set, use that in preference.
+    if (const char* env_path = std::getenv("NMC_DATADIR")) {
+        return util::path(env_path);
```

Comment:
I think that is reasonable, there just needs to be a little bit of documentation that explains that the path that is returned may not be valid.